### PR TITLE
[15.0][MIG] document_page_approval: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @ernestotejeda can you review it?

@Tecnativa TT36950

### CODE REVIEW

File: document_page_approval/models/document_page.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+class DocumentPage(models.Model):
+    """Useful to know the state of a document."""
+
+    _inherit = "document.page"
+
+    history_ids = fields.One2many(
+        order="approved_date DESC", domain=[("state", "=", "approved")]
+    )
```
Comment (FernandoRomera) [inline document_page_approval/models/document_page.py]: _order="approved_date DESC"

history_ids = fields.One2many(
         domain=[("state", "=", "approved")]
    )

Comment (victoralmau) [reply document_page_approval/models/document_page.py]: `order` or `_order` attribute does not work on a field.
I have added this ordering to the tree view with `default_order` attribute.

Comment (FernandoRomera) [reply document_page_approval/models/document_page.py]: I think there was a mistake. To correct: remove the field order attribute and add the _order attribute at the table level.

Comment (pedrobaeza) [reply document_page_approval/models/document_page.py]: But that's not the intention. The goal is to order the children model, so the other technique is the required one.

File: document_page_approval/models/document_page.py [hunk 2]
```diff
@@ -48,6 +46,7 @@ class DocumentPage(models.Model):
         "Approval required",
         help="If true, changes of this page require approval",
         compute="_compute_is_approval_required",
+        recursive=True,
```
Comment (pedrobaeza) [inline document_page_approval/models/document_page.py:49]: Why do you need this?

File: document_page_approval/models/document_page.py [hunk 3]
```diff
@@ -57,17 +56,20 @@ class DocumentPage(models.Model):
         string="Approver groups",
         help="Groups that can approve changes to this document",
         compute="_compute_approver_group_ids",
+        recursive=True,
```
Comment (pedrobaeza) [inline document_page_approval/models/document_page.py:59]: And this?

File: document_page_approval/models/document_page.py [hunk 4]
```diff
@@ -57,17 +56,20 @@ class DocumentPage(models.Model):
...
     )
 
     has_changes_pending_approval = fields.Boolean(
         compute="_compute_has_changes_pending_approval",
-        string="Has changes pending approval",
```
Comment (pedrobaeza) [inline document_page_approval/models/document_page.py]: Isn't this changing the casing of the string to `Has Changes Pending Approval`?

File: document_page_approval/models/document_page_history.py [hunk 5]
```diff
@@ -11,6 +11,7 @@ class DocumentPageHistory(models.Model):
 
     _name = "document.page.history"
     _inherit = ["document.page.history", "mail.thread"]
+    _order = "id DESC"
```
Comment (pedrobaeza) [inline document_page_approval/models/document_page_history.py]: Shouldn't this be ordered by `approved_date`?

File: document_page_approval/models/document_page_history.py [hunk 6]
```diff
@@ -126,8 +127,8 @@ def action_cancel(self):
         for rec in self:
             rec.message_post(
                 subtype_xmlid="mail.mt_comment",
-                body=_("Change request <b>%s</b> has been cancelled by %s.")
-                % (rec.display_name, self.env.user.name),
+                body=_("Change request <b>%(name)s</b> has been cancelled by %(user)s.")
```
Comment (pedrobaeza) [inline document_page_approval/models/document_page_history.py:130]: Change translations for this, at least on Spanish

File: document_page_approval/models/document_page.py (hunk 2 above)
Comment (victoralmau) [reply document_page_approval/models/document_page.py:49]: Without this change, when executing the tests, the following log is shown:

`WARNING devel py.warnings: /opt/odoo/custom/src/odoo/odoo/fields.py:722: UserWarning: Field document.page.approver_group_ids should be declared with recursive=True`

File: document_page_approval/models/document_page.py (hunk 4 above)
Comment (victoralmau) [reply document_page_approval/models/document_page.py]: Without this change, when executing pre-commit the following log is shown:

`The attribute string is redundant. String parameter equal to name of variable`

Comment (pedrobaeza) [reply document_page_approval/models/document_page.py]: Yeah, but that's not my question? If you remove it, what is the label? If it's with capital letters, then the pylint check is not correct. If it's the same, then my bad and everything is correct.

File: document_page_approval/models/document_page.py (hunk 2 above)
Comment (pedrobaeza) [reply document_page_approval/models/document_page.py:49]: Uhm, OK, but I don't know why

File: document_page_approval/models/document_page.py (hunk 4 above)
Comment (victoralmau) [reply document_page_approval/models/document_page.py]: Ok sorry, removing the string tag odoo auto-defines it with "_Has Changes Pending Approval_" (wrong). The pylint validation seems to be wrong.

Comment (pedrobaeza) [reply document_page_approval/models/document_page.py]: Please put the issue in pylint-odoo project.

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @ernestotejeda

**Comment by victoralmau** [comment]
Ping @chienandalu

**Review by chienandalu — APPROVED** [review]
Tested :+1:

**Pull request merged**